### ISSUE TITLE
Issue when processing too much data

### ISSUE THREAD

**Issue by jiayu2B** [pyannote/pyannote-database#24]
Hi, I found an issue when using pyannote with database I created. The thing is that the training speed is too slow, nearly 10s/batch, and impossible to train. 
I think the reason is that I used my database, which contains 5 million .npy files ( 900 GB), 30 million lines of .mdtm and 5 million lines of .uem files. Perhaps GPU cannot hold that much, and it use CPU instead.
When I use 5 thousand .npy files, this problem disappeared, and the training speed is 10 batchs/s.
Do you know the reason why this issue occurs? And how can I fix it? 
Thank you!

**Comment by hbredin** [comment]
Can you please share your `config.yml`?

**Comment by jiayu2B** [comment]
> task:
>    name: SpeechActivityDetection
>    params:
>       duration: 2      # sequences are 2s long
>       batch_size: 64     # 64 sequences per batch
>       per_epoch: 1       # one epoch = 1 day of audio
> 
> #feature_extraction:
> #   name: LibrosaMFCC      # use MFCC from librosa
> #   params:
> #      e: False            # do not use energy
> #      De: True            # use energy 1st derivative
> #      DDe: True           # use energy 2nd derivative
> #      coefs: 19           # use 19 MFCC coefficients
> #      D: True             # use coefficients 1st derivative
> #      DD: True            # use coefficients 2nd derivative
> #      duration: 0.025     # extract MFCC from 25ms windows
> #      step: 0.010         # extract MFCC every 10ms
> #      sample_rate: 16000  # convert to 16KHz first (if needed)
> feature_extraction:
>    name: Precomputed
>    params:
>       root_dir: tutorials/feature_extraction
> 
> architecture:
>    name: StackedRNN
>    params:
>       instance_normalize: True  # normalize sequences
>       rnn: LSTM                 # use LSTM (could be GRU)
>       recurrent: [128, 128]     # two layers with 128 hidden states
>       bidirectional: True       # bidirectional LSTMs
>       linear: [32, 32]          # add two linear layers at the end
> 
> scheduler:
>    name: CyclicScheduler        # use cyclic learning rate (LR) scheduler
>    params:
>       learning_rate: auto       # automatically guess LR upper bound
>       epochs_per_cycle: 14      # 14 epochs per cycle
> 
>

**Comment by jiayu2B** [comment]
task:
   name: SpeechActivityDetection
   params:
      duration: 2      # sequences are 2s long
      batch_size: 64     # 64 sequences per batch
      per_epoch: 1       # one epoch = 1 day of audio
feature_extraction:
   name: Precomputed
   params:
      root_dir: tutorials/feature_extraction

architecture:
   name: StackedRNN
   params:
      instance_normalize: True  # normalize sequences
      rnn: LSTM                 # use LSTM (could be GRU)
      recurrent: [128, 128]     # two layers with 128 hidden states
      bidirectional: True       # bidirectional LSTMs
      linear: [32, 32]          # add two linear layers at the end

scheduler:
   name: CyclicScheduler        # use cyclic learning rate (LR) scheduler
   params:
      learning_rate: auto       # automatically guess LR upper bound
      epochs_per_cycle: 14      # 14 epochs per cycle

**Comment by hbredin** [comment]
Try using the following configuration for the feature extraction part.
You might have too many open files at the same time leading to this decrease in performance.

```yaml
feature_extraction:
    name: Precomputed
    params:
        root_dir: tutorials/feature_extraction
        use_memmap: False
```

Please let me know if that solves your issue.
